cloud-storage/googledrive/impl/src/java/org/sakaiproject/googledrive/service/GoogleDriveServiceImpl.java

```diff
@@ -0,0 +1,359 @@
+			((ObjectNode) webInfo).put("client_secret", clientSecret);
+			((ObjectNode) rootNode).set("web", webInfo);
+			InputStream isJson = IOUtils.toInputStream(rootNode.toString());
+			isJson.close();//?
+
+			httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+			jsonFactory = JacksonFactory.getDefaultInstance();
+			clientSecrets = GoogleClientSecrets.load(jsonFactory, new InputStreamReader(isJson));
+
+			DataStoreFactory dataStore = new JPADataStoreFactory(googledriveRepo);			
+			flow = new GoogleAuthorizationCodeFlow.Builder(httpTransport, jsonFactory, clientSecrets, Collections.singleton(DriveScopes.DRIVE))
+					.setDataStoreFactory(dataStore)
+					.setApprovalPrompt("force")
+					.setAccessType("offline")
+					.build();
+		} catch(Exception e) {
+			log.error("Error while trying to init Google Drive configuration");
+		}
+
+		driveRootItemsCache = memoryService.<String, List<GoogleDriveItem>>getCache("org.sakaiproject.googledrive.service.driveRootItemsCache");
+		driveChildrenItemsCache = memoryService.<String, List<GoogleDriveItem>>getCache("org.sakaiproject.googledrive.service.driveChildrenItemsCache");
+		googledriveUserCache = memoryService.<String, Drive>getCache("org.sakaiproject.googledrive.service.googledriveUserCache");
+	}
+
+	public String formAuthenticationUrl() {
+		log.debug("formAuthenticationUrl");
+		if(!isConfigured()){
+			return null;
+		}
+
+		String redirectTo = flow.newAuthorizationUrl()
+				.setRedirectUri(redirectUri)
+				.setState(redirectUri)
+				.build();
+
+		log.debug("redirectTo : {}", redirectTo);
+		return redirectTo;
+	}
+
+	public boolean token(String userId, String code) {
+		if (!isConfigured()) {
+			return false;
+		}
+
+		try {
+			GoogleTokenResponse googleResponse = flow.newTokenRequest(code)
+					.setRedirectUri(redirectUri)
+					.execute();
+			Credential cred = flow.createAndStoreCredential(googleResponse, userId);
+
+			service = new Drive.Builder(httpTransport, jsonFactory, cred)
+				.setApplicationName(GOOGLEDRIVE_APP_NAME)
+				.build();
+
+			Drive.About about = service.about();
+			Drive.About.Get get = about.get().setFields("user(displayName, emailAddress, permissionId)");
+			About ab = get.execute();			
+			log.debug("About : {}", ab.toString());
+			
+			GoogleDriveUser du = getGoogleDriveUser(userId);
+			du.setGoogleDriveUserId(ab.getUser().getPermissionId());
+			du.setGoogleDriveName(ab.getUser().getEmailAddress());
+			googledriveRepo.update(du);
+			return true;
+
+		} catch(Exception e) {
+			log.warn("GoogleDrive: Error while retrieving or saving the credentials for user {} : {}", userId, e.getMessage());
+			revokeGoogleDriveConfiguration(userId);
+		}
+		return false;
+	}
+
+	public Drive refreshToken(String userId){
+		if(!isConfigured()){
+			return null;
+		}
+		Drive googledriveUser = googledriveUserCache.get(userId);
+		if(googledriveUser != null) {
+			log.debug("refreshToken : Reusing previous user data " + googledriveUser);
```

Comment:
You should probably guard this debug call.

Reply:
Guard is only necessary when using string concatenation, like is happening here. If you use the new logger style with replacement values, guarding is not necessary as the string concat doesn't happen regardless of logger level. In a nutshell, guard this here or convert it to the new style.

Reply:
New style much cleaner:

log.debug("refreshToken : Reusing previous user data {}", googledriveUser);